### PR TITLE
feat: Introduce new node type: embedded inline resource [DANTE-1167] 

### PR DESCRIPTION
**Pull request by msieroslawska**
This PR is the **second** part of introducing embedded inline cross-space links, following the [initial refactoring](https://github.com/contentful/field-editors/pull/1516). It introduces the logic of inserting embedded inline entries.

Not covered in this PR (will come as a follow up):
- support for pasting from clipboard
- cleanup of components (reducing code repetition, eg. `EmbeddedInlineToolbarIcon` and `EmbeddedBlockToolbarIcon`)

Some screenshots:
Screenshot 2023-10-16 at 10 06 59
Screenshot 2023-10-16 at 10 07 07
Screenshot 2023-10-16 at 10 07 18
Screenshot 2023-10-16 at 10 07 24

### CODE REVIEW

File: packages/rich-text/src/plugins/shared/EmbeddedInlineToolbarIcon.tsx [hunk 1]
```diff
@@ -59,7 +70,10 @@ export function EmbeddedInlineToolbarIcon(props: EmbeddedInlineToolbarIconProps)
           variant="secondary"
           className={`rich-text__embedded-entry-list-icon ${styles.icon}`}
         />
-        <span>Inline entry</span>
+        <span>
+          Inline entry
+          {nodeType == INLINES.EMBEDDED_RESOURCE && <ResourceNewBadge />}
```
Comment (chrishelgert) [inline packages/rich-text/src/plugins/shared/EmbeddedInlineToolbarIcon.tsx:75]:
```suggestion
          {nodeType === INLINES.EMBEDDED_RESOURCE && <ResourceNewBadge />}
```

;)

**Comment by chrishelgert** [comment]
Please consider adding the tests directly to this PR, the editor is always vulnerable to bugs :(
And we don't want to break your feature on mistake :)

**Comment by msieroslawska** [comment]
@chrishelgert I have added an e2e scenario covering the new node. It did uncover an issue we had with the schema, thanks for pushing for adding tests! Could you take another look please?

**Review by chrishelgert — APPROVED** [review]
LGTM 👍

**Pull request merged**